src/main/java/Report.java

```diff
@@ -0,0 +1,129 @@
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Report class retrieves information about students from List<Student>
+ * and is able to generate reports based on this information.
+ */
+public class Report {
+    private final List<Student> students;
+
+    public Report(List<Student> students) {
+        this.students = students;
+    }
+
+    /**
+     * generateShortReport returns short, sketchy information - one line per student.
+     */
+    public void generateShortReport() {
+
+        for (Student student : students) {
+            System.out.printf("%s(%s) - ", student.getName(), student.getCurriculum());
+
+            int programDuration = 0;
+            for (Map.Entry<String, Integer> entry : student.getCourseMap().entrySet()) {
+                programDuration += entry.getValue();
+            }
+
+            int days;
+            if (TrainingTimeManager.isTrainingFinished(student.getStartDateTime(), programDuration)) {
+                System.out.print("Training completed - ");
+                int hoursAfterCompletion = (TrainingTimeManager.calculateHoursToTrainingCompletion(
+                        student.getStartDateTime(), programDuration));
+
+                if (hoursAfterCompletion < 8) {
+                    System.out.printf("Time passed since completion: %dhrs\n", hoursAfterCompletion);
+                } else {
+                    days = 0;
+                    while (hoursAfterCompletion >= 8) {
+                        days++;
+                        hoursAfterCompletion -= 8;
+                    }
+                    System.out.printf("Time passed since completion: %dd %dhrs\n", days, hoursAfterCompletion);
+                }
+
+            } else {
+                System.out.print("Training still in progress - ");
+                int hoursToCompletion = TrainingTimeManager.calculateHoursToTrainingCompletion(
+                        student.getStartDateTime(), programDuration);
+
+                if (hoursToCompletion < 8) {
+                    System.out.printf("Time left to completion: %dhrs\n", hoursToCompletion);
+                } else {
+                    days = 0;
```

Comment:
0, 1 are already available in Apache Commons `NumberUtils` as `INTEGER_ZERO` / `LONG_ONE` / etc. You can try to add Apache commons to the project and start checking what is available in it. It provides a great collection of NULL safe methods to work with Strings/Numbers/etc.